### ISSUE TITLE
Pointer being freed was not allocated

### ISSUE THREAD

**Issue by faizshukri** [camme/node-nfc#17]
After compile the code, I try to run the test file. However, I got an error about pointer being freed was not allocated.

Here the full output.

```bash
version: { name: 'libnfc', version: '1.7.0' }
devices: { 'acr122_pcsc:ACS ACR122U':
   { name: 'ACS ACR122U / ACR122U214',
     info:
      { chip: 'PN532 v1.6',
        'initator mode modulations':
         { 'ISO/IEC 14443A': [ '106 kbps' ],
           FeliCa: [ '424 kbps', '212 kbps' ],
           'ISO/IEC 14443-4B': [ '106 kbps' ],
           'Innovision Jewel': [ '106 kbps' ],
           'D.E.P.': [ '424 kbps', '212 kbps', '106 kbps' ] },
        'target mode modulations':
         { 'ISO/IEC 14443A': [ '106 kbps' ],
           FeliCa: [ '424 kbps', '212 kbps' ],
           'D.E.P.': [ '424 kbps', '212 kbps', '106 kbps' ] } } } }

{ deviceID: 'acr122_pcsc:ACS ACR122U',
  name: 'ACS ACR122U / ACR122U214' }
{ deviceID: 'acr122_pcsc:ACS ACR122U',
  name: 'ACS ACR122U / ACR122U214',
  uid: '04:99:84:6a:43:2f:80',
  type: 68,
  tag: 'mifare-ultralight',
  data: <Buffer 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 03 00 84 01 01 00 00 00 50 5b 84 01 01 00 00 00 02 00 52 31 32 32 55 00 80 02 16 10 00 00 07 00 01 00 ... >,
  offset: 16 }
[]
stopped
node(28605,0x7fff790d2000) malloc: *** error for object 0x101600b20: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
[1]    28605 abort      node test.js
```

Reader: ACR122U
Tag: Mifare Ultralight
OS: OSX El Capitan
Node: 4.2.1

**Comment by faizshukri** [comment]
I try with Node v0.10 and v0.12, both work fine. The error occurs for v4

**Comment by jeroenvollenbrock** [comment]
fixed in #18

**Comment by jonas-lauber** [comment]
Hi everybody.

I am having the exact same error in another node's project wrapping C code too. Could anyone give me hints on how to solve this issue ? 

Your help would be greatly appreciated, because I have no idea how to figure that out... :-(
